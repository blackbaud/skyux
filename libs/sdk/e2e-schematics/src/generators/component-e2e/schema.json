{
  "$schema": "http://json-schema.org/schema",
  "$id": "library-e2e",
  "title": "Create a library e2e",
  "description": "Creates an e2e infrastructure for a library.",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "The name of the library.",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What is the name of the library?",
      "pattern": "^[a-zA-Z].*$"
    },
    "tags": {
      "type": "string",
      "description": "Add tags to the library (used for linting)."
    }
  },
  "additionalProperties": false,
  "required": ["name"]
}
