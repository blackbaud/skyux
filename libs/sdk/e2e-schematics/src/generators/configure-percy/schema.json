{
  "$schema": "http://json-schema.org/schema",
  "$id": "configure-percy",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "e2e project name",
      "$default": {
        "$source": "argv",
        "index": 0
      }
    }
  }
}
