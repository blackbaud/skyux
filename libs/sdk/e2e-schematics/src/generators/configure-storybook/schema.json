{
  "$schema": "http://json-schema.org/schema",
  "$id": "configure-storybook",
  "title": "Configure Storybook",
  "description": "Adds configuration to Storybook.",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "The name of the project using Storybook.",
      "$default": {
        "$source": "argv",
        "index": 0
      }
    }
  },
  "additionalProperties": false
}
