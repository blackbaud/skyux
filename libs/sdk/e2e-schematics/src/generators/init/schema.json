{
  "$schema": "http://json-schema.org/schema",
  "cli": "nx",
  "$id": "Init",
  "title": "Configure E2E environment"
}
