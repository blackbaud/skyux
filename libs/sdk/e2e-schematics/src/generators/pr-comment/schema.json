{
  "$schema": "http://json-schema.org/schema",
  "$id": "PrComment",
  "title": "PR Comment",
  "description": "Generate markdown for a PR comment or a PR preview README.md",
  "type": "object",
  "properties": {
    "pr": {
      "type": "number",
      "description": "PR Number",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "Which PR is this for?"
    },
    "storybooks": {
      "type": "string",
      "description": "Comma separated list of storybooks to include in the comment",
      "alias": "s",
      "x-prompt": "Which storybook projects should be listed?"
    },
    "apps": {
      "type": "string",
      "description": "Comma separated list of apps to include in the comment",
      "alias": "a"
    }
  },
  "required": ["pr", "storybooks"]
}
