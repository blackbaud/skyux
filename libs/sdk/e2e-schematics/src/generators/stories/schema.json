{
  "$schema": "http://json-schema.org/schema",
  "$id": "Stories",
  "title": "Create Storybook stories/specs",
  "description": "Creates Storybook stories/specs for all Angular components declared in a project.",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What's the name of the project for which you want to generate stories?",
      "aliases": [
        "app",
        "storybook-app",
        "storybookapp",
        "storybookApp",
        "storybook"
      ]
    },
    "generateCypressSpecs": {
      "type": "boolean",
      "description": "Specifies whether to automatically generate `*.spec.ts` files in the Cypress e2e app generated by the `cypress-configure` generator.",
      "default": true
    },
    "cypressProject": {
      "type": "string",
      "description": "The Cypress project to generate the stories under. This is inferred from `name` by default."
    }
  },
  "additionalProperties": false,
  "required": ["project"]
}
