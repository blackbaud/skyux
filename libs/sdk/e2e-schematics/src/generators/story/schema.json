{
  "$schema": "http://json-schema.org/schema",
  "$id": "Story",
  "title": "Create a new story with a new component and corresponding tests",
  "description": "Create a new story with a new component and corresponding tests",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "The name of the component.",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use for the new storybook component?"
    },
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "x-prompt": "What's the name of the storybook application for which you want to generate the story?",
      "aliases": [
        "app",
        "storybook-app",
        "storybookapp",
        "storybookApp",
        "storybook"
      ]
    },
    "generateCypressSpecs": {
      "type": "boolean",
      "description": "Specifies whether to automatically generate `*.spec.ts` files in the Cypress e2e app generated by the `cypress-configure` generator.",
      "default": true
    },
    "cypressProject": {
      "type": "string",
      "description": "The Cypress project to generate the stories under. This is inferred from `project` by default."
    },
    "includeTests": {
      "type": "boolean",
      "description": "Include tests in the library.",
      "default": false
    }
  },
  "additionalProperties": false,
  "required": ["name", "project"]
}
