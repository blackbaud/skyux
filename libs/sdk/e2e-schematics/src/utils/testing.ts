import {
  E2eTestRunner,
  UnitTestRunner,
  applicationGenerator,
  libraryGenerator,
} from '@nx/angular/generators';
import { Tree } from '@nx/devkit';
import { Linter } from '@nx/eslint';

export async function createTestApplication(
  tree: Tree,
  options: { name: string; e2eTestRunner?: boolean; unitTestRunner?: boolean },
): Promise<void> {
  await applicationGenerator(tree, {
    name: options.name,
    skipPackageJson: true,
    directory: `apps/${options.name}`,
    e2eTestRunner: options.e2eTestRunner
      ? E2eTestRunner.Cypress
      : E2eTestRunner.None,
    linter: Linter.None,
    unitTestRunner: options.unitTestRunner
      ? UnitTestRunner.Jest
      : UnitTestRunner.None,
  });
}

export async function createTestLibrary(
  tree: Tree,
  options: { name: string },
): Promise<void> {
  await libraryGenerator(tree, {
    name: options.name,
    skipPackageJson: true,
    directory: `libs/${options.name}`,
    linter: Linter.None,
    unitTestRunner: UnitTestRunner.None,
  });
}
