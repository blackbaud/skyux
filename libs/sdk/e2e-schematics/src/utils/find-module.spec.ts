import { applicationGenerator } from '@nx/angular/generators';
import { readProjectConfiguration } from '@nx/devkit';
import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';
import { RoutingScope } from '@schematics/angular/module/schema';

import {
  angularComponentGenerator,
  angularModuleGenerator,
} from './angular-module-generator';
import {
  applyTransformersToPath,
  findNgModuleClass,
  getInsertImportTransformer,
  readSourceFile,
} from './ast-utils';
import {
  findClosestModule,
  findDeclaringModule,
  findModulePaths,
  isRoutingModule,
} from './find-module';

describe('find-module', () => {
  it('should find module', async () => {
    const tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
    await applicationGenerator(tree, {
      name: 'test',
    });
    expect(readProjectConfiguration(tree, 'test').sourceRoot).toBe(
      'apps/test/src'
    );
    await angularModuleGenerator(tree, { name: 'test', project: 'test' });
    await angularComponentGenerator(tree, {
      name: 'test',
      project: 'test',
      module: 'test',
    });
    expect(tree.listChanges().map((c) => c.path)).toMatchSnapshot();
    expect(
      tree.read('apps/test/src/app/test/test.module.ts', 'utf-8')
    ).toMatchSnapshot();
    const module = findDeclaringModule(
      tree,
      'apps/test/src',
      'apps/test/src/app/test/test.component.ts'
    );
    expect(module?.filepath).toBe('apps/test/src/app/test/test.module.ts');
    expect(module?.module.classDeclaration.name?.text).toBe('TestModule');
  });

  it('should find module, skip non-declaring reference', async () => {
    const tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
    await applicationGenerator(tree, {
      name: 'test',
    });
    await angularModuleGenerator(tree, {
      name: 'test/bogus',
      flat: true,
      project: 'test',
    });
    applyTransformersToPath(tree, 'apps/test/src/app/test/bogus.module.ts', [
      getInsertImportTransformer('TestComponent', './test.component'),
    ]);
    await angularModuleGenerator(tree, { name: 'test', project: 'test' });
    await angularComponentGenerator(tree, {
      name: 'test',
      project: 'test',
      module: 'test',
    });
    const module = findDeclaringModule(
      tree,
      'apps/test/src',
      'apps/test/src/app/test/test.component.ts'
    );
    expect(module?.filepath).toBe('apps/test/src/app/test/test.module.ts');
    expect(module?.module.classDeclaration.name?.text).toBe('TestModule');
  });

  it('should find routing module', async () => {
    const tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
    await applicationGenerator(tree, {
      name: 'test',
    });
    await angularModuleGenerator(tree, {
      name: 'test',
      project: 'test',
      routing: true,
      routingScope: RoutingScope.Root,
    });
    await angularComponentGenerator(tree, {
      name: 'test',
      project: 'test',
      module: 'test',
    });
    const modules = findModulePaths(
      tree,
      'apps/test/src',
      (path) => {
        return (
          tree
            .read(path, 'utf-8')
            ?.includes('export class TestRoutingModule ') ?? false
        );
      },
      '-routing.module.ts'
    );
    expect(modules).toEqual(['apps/test/src/app/test/test-routing.module.ts']);
  });

  it('should isRoutingModule', async () => {
    const tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
    await applicationGenerator(tree, {
      name: 'test',
    });
    await angularModuleGenerator(tree, {
      name: 'test',
      project: 'test',
      routing: true,
      routingScope: RoutingScope.Root,
    });
    await angularModuleGenerator(tree, {
      name: 'sub-test',
      project: 'test',
      routing: true,
      routingScope: RoutingScope.Child,
      route: 'test',
      module: 'test',
    });
    await angularComponentGenerator(tree, {
      name: 'test',
      project: 'test',
      module: 'test',
    });
    const modules = findModulePaths(tree, 'apps/test/src', (path) => {
      const source = readSourceFile(tree, path);
      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
      const module = findNgModuleClass(source)!;
      return !!module && isRoutingModule(module, source);
    });
    expect(modules).toEqual([
      'apps/test/src/app/test/test-routing.module.ts',
      'apps/test/src/app/sub-test/sub-test-routing.module.ts',
    ]);
  });

  it('should findClosestModule', async () => {
    const moduleList = [
      'apps/test/src/app/best/best-routing.module.ts',
      'apps/test/src/app/test/test-routing.module.ts',
      'apps/test/src/app/rest/rest-routing.module.ts',
      'apps/test/src/app/test/sub/sub-routing.module.ts',
      'apps/test/src/app/app-routing.module.ts',
      'apps/test/src/app/something/else/complex/complex-routing.module.ts',
      'apps/test/src/app/complex/something/else/else-routing.module.ts',
      'apps/test/src/app/something/something-routing.module.ts',
    ];
    expect(findClosestModule(moduleList, 'apps/test/src', 'app/other')).toBe(
      'app-routing'
    );
    expect(
      findClosestModule(moduleList, 'apps/test/src', 'app/test/sub/deep')
    ).toBe('sub-routing');
    expect(findClosestModule(moduleList, 'apps/test/src', 'app/test/sub')).toBe(
      'sub-routing'
    );
    expect(findClosestModule(moduleList, 'apps/test/src', 'app/test')).toBe(
      'test-routing'
    );
    expect(findClosestModule(moduleList, 'apps/test/src', 'app')).toBe(
      'app-routing'
    );
    expect(
      findClosestModule(moduleList, 'apps/test/src', 'outside')
    ).toBeFalsy();
    expect(findClosestModule(moduleList, 'apps/other/src', 'app')).toBeFalsy();
  });
});
