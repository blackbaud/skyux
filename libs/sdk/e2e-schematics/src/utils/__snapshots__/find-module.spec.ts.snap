// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`find-module should find module 1`] = `
Array [
  ".prettierrc",
  "package.json",
  "nx.json",
  "tsconfig.base.json",
  "apps/.gitignore",
  "libs/.gitignore",
  ".prettierignore",
  "jest.preset.js",
  "jest.config.ts",
  "apps/test/project.json",
  "apps/test/src/assets/.gitkeep",
  "apps/test/src/favicon.ico",
  "apps/test/src/index.html",
  "apps/test/src/styles.css",
  "apps/test/tsconfig.app.json",
  "apps/test/tsconfig.editor.json",
  "apps/test/tsconfig.json",
  "apps/test/src/app/app.component.css",
  "apps/test/src/app/app.component.html",
  "apps/test/src/app/app.component.spec.ts",
  "apps/test/src/app/app.component.ts",
  "apps/test/src/app/app.module.ts",
  "apps/test/src/app/nx-welcome.component.ts",
  "apps/test/src/main.ts",
  ".eslintrc.json",
  ".eslintignore",
  "apps/test/.eslintrc.json",
  "apps/test/jest.config.ts",
  "apps/test/src/test-setup.ts",
  "apps/test/tsconfig.spec.json",
  "apps/test-e2e/cypress.config.ts",
  "apps/test-e2e/src/e2e/app.cy.ts",
  "apps/test-e2e/src/fixtures/example.json",
  "apps/test-e2e/src/support/app.po.ts",
  "apps/test-e2e/src/support/commands.ts",
  "apps/test-e2e/src/support/e2e.ts",
  "apps/test-e2e/tsconfig.json",
  "apps/test-e2e/project.json",
  "apps/test-e2e/.eslintrc.json",
  "apps/test/src/app/test/test.module.ts",
  "apps/test/src/app/test/test.component.css",
  "apps/test/src/app/test/test.component.html",
  "apps/test/src/app/test/test.component.spec.ts",
  "apps/test/src/app/test/test.component.ts",
]
`;

exports[`find-module should find module 2`] = `
"import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { TestComponent } from './test.component';



@NgModule({
  declarations: [
    TestComponent
  ],
  imports: [
    CommonModule
  ]
})
export class TestModule { }
"
`;
